### PR TITLE
Two pointer solution for Q1 and 16

### PR DESCRIPTION
**Pull request by xiaqian1999**

### CODE REVIEW

File: xiaqianZhang/assignments/twopointers/lc1/lc1.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#     if currentSum == target_sum:
+#       return [leftIndex, rightIndex]
+
+#     elif currentSum < target_sum:
+#       leftIndex +=1
+    
+#     else:
+#       rightIndex -=1
+
+#   return []
```
Comment (knrodriguez) [inline xiaqianZhang/assignments/twopointers/lc1/lc1.py:33]: Why is your code commented out?

File: xiaqianZhang/assignments/twopointers/lc1/lc1.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# output: pair of index that the numbers added up to the target sum
+# edge cases: empty array, sum not found -> return empty array
+# left index that initialize starting at arr[0]
+# right index that initialize at the end of arr = len(arr)-1
+# loop through the array while left integer < right integer
+# currentSum = left int + right int
+#   check if currentSum == targetSum Yes-> return the index where both int located
+#   if currentSum < targetSum -> increment the left index
+#   if currentSum > targetSum -> decrement the right index
+# if loop through the entire list and can't found any index that added up to target Sum, return []
```
Comment (knrodriguez) [inline xiaqianZhang/assignments/twopointers/lc1/lc1.py:16]: Love this approach!

File: xiaqianZhang/assignments/twopointers/lc1/lc1.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# Problem Statement #
+# Given an array of sorted numbers and a target sum, find a pair in the array whose sum is equal to the given target.
+
+# Write a function to return the indices of the two numbers (i.e. the pair) such that they add up to the given target.
+
+# input: sorted list that may including negative numbers, target sum that may be negative
+# output: pair of index that the numbers added up to the target sum
+# edge cases: empty array, sum not found -> return empty array
```
Comment (knrodriguez) [inline xiaqianZhang/assignments/twopointers/lc1/lc1.py:8]: Great details on your inputs, outputs, and edge cases!

File: xiaqianZhang/assignments/twopointers/lc16/lc16.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
+# Problem Statement #
```
Comment (knrodriguez) [inline xiaqianZhang/assignments/twopointers/lc16/lc16.py:1]: This code does not run, unless you put the `import math` statement on top.

File: xiaqianZhang/assignments/twopointers/lc16/lc16.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+# input: array of unsorted numbers that includes negative integers, target sum might also be negative
+# output: return the sum of three integers that is closest to the target sum, prefer result sum <= target sum than >= target sum
+# first by sort the array 
+# then initialize a smallest difference variable as larger numbers such as infinite number
+# Loop through the entire arr
+# initialize three pointer as a = first index of the array, b = a+1 c = len(arr)-1, which is the last index of the arr 
+# Then initialize variable current different that we want to constant update
+# if the smallestDiff = 0, then that means the result sum is == target sum
+# otherwise we want to continue loop through the entire arr and find smaller difference as possible
+# edge cases: if the length of the arr is less than the 3, then we simply return the sum of all the int in the given arr
```
Comment (knrodriguez) [inline xiaqianZhang/assignments/twopointers/lc16/lc16.py:30]: Love that you wrote out the 2 approaches!

File: xiaqianZhang/assignments/twopointers/lc75/lc75.py [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+# output: return the sorted array without using sorted function and in O(n) times
+# when we sorted, the order is from 0, 1, 2. Where all the 0s are on the left hand side, and all the 2s are on the right hand side
+# initialize the lowerPointer = 0 [most left hand side]
+# initialize the higherPointer = len(arr)-1 [most right hand side]
+# When we loop through the array
+# if the variable == 0: we want to move it to the left hand side, swap lowerPointer = variable, 
+# after swap, we increment i+=1, increment lowerPointer+=1 since the current lowerPointer is 0 as we expected 
+# if the variable == 1: we will ignore and increment index+=1
+# if the variable == 2: swap the higherPointer = variable, decrement higherPointer-=1 since the current higherPointer is 2 as we expected
+# edge cases: if the arr is empty/only 1 variable, we could just return the array
```
Comment (knrodriguez) [inline xiaqianZhang/assignments/twopointers/lc75/lc75.py:61]: It is wonderful that you took the time to refactor your solution to make it more optimized! Your approach is detailed, you have included edge cases and test cases to match. Nice work :100:

**Review by knrodriguez — APPROVED** [review]
